libs/gltfio/materials/ubershader.mat.in

```diff
@@ -75,13 +80,48 @@ fragment {
             material.metallic = materialParams.metallicFactor;
             material.emissive.rgb = materialParams.emissiveFactor.rgb;
             material.emissive.a = 3.0;
+
+            float glossiness = materialParams.glossinessFactor;
+
             if (materialParams.metallicRoughnessIndex > -1) {
                 float2 uv = uvs[materialParams.metallicRoughnessIndex];
                 uv = (vec3(uv, 1.0) * materialParams.metallicRoughnessUvMatrix).xy;
-                vec4 roughness = texture(materialParams_metallicRoughnessMap, uv);
-                material.roughness *= roughness.g;
-                material.metallic *= roughness.b;
+
+                if (materialParams.useSpecularGlossiness == 1) {
+                    vec4 sg = texture(materialParams_metallicRoughnessMap, uv);
+                    glossiness *= sg.a;
+                    vec3 specColor = sg.rgb * materialParams.specularFactor;
+
+                    // NOTE: this conversion logic is a carry-over from MeshAssimp. According to the
+                    // spec: "The specular property from specular-glossiness material model is the
+                    // same as the base color value from the metallic-roughness material model for
+                    // metals."
+                    material.reflectance = 0.5;
+                    material.metallic = 0.0;
+                    if (specColor.r != specColor.g && specColor.r != specColor.b) {
```

Comment:
That would merit comparing against the spec or revisiting. I'm not sure how correct this is

Reply:
For starters I think we should simplify this by removing the weird conditionals and just say: "material.baseColor.rgb = specColor" unconditionally.  Many models seem look to reasonable when I tried this.  Need to think more about handling their concept of "diffuse color" correctly though, that's tricky...

Reply:
Doesn't it mean that everything using the specularGlossiness model would end up being a metal? I imagine a plastic in that model has a white specular RGB color and a colored diffuse color.

I thought about this some more and as much as I hate it, the only proper solution is to add a new material model (and we should discourage its usage). This model would have `baseColor` for `diffuse` and add a specularColor (or reuse sheenColor?) attribute. `metallic` and `reflectance` would go away.